implementations/rust/ockam/ockam_node/src/storage/database/migrations/migration_support/migrator.rs

```diff
@@ -67,29 +67,11 @@ impl Migrator {
 }
 
 impl Migrator {
-    fn is_inside_version_range(
-        version: i64,
-        from_version: i64, // including
-        to_version: i64,
-        including_to: bool,
-    ) -> bool {
-        if from_version <= version && version < to_version {
-            return true;
-        }
-
-        if version == to_version && including_to {
-            return true;
-        }
-
-        false
-    }
-
     async fn run_migrations(
         &self,
         connection: &mut SqliteConnection,
-        from_version: i64,  // including
-        to_version: i64,    // not including
-        run_last_sql: bool, // will run sql migration with verion == to_version
+        version_interval: VersionInterval,
+        run_last_sql_migration: bool,
```

Comment:
Why not extend the `VersionInterval` data type so that either:
 
 - It has a function `set_last_version_inclusive(self) -> Self`
 - It has  2 constructors `from_to` and `from_to_inclusive`

This way we directly decide if the last version should be in when building the range.

Reply:
I didn't do that because of how it's used later: see lines 87-105. I'm not entirely sure why we treat rust/sql differently in that block.

Reply:
I think it's because how it's used in tests. I'm going to try to change a little how we use it so we can define the interval properly.